### ISSUE TITLE
Map key heuristics with constant fields

### ISSUE THREAD

**Issue by GoogleCodeExporter** [typetools/checker-framework#388]
```
The nullness checker reports an incorrect error for the following code.

This would likely be resolved by the fix for Issue 221.

I tested with javac 1.8.0-jsr308-1.8.9.

===
import java.util.Map;

class MapKeyConstant {
  static class Holder {
    static final String KEY = "key";
  }

  public String getOrDefault(Map<String, String> map, String defaultValue) {
    if (map.containsKey(Holder.KEY)) {
      return map.get(Holder.KEY);
    } else {
      return defaultValue;
    }
  }
}
===

Test.java:10: error: [return.type.incompatible] incompatible types in return.
      return map.get(Holder.KEY);
                    ^
  found   : @Initialized @Nullable String
  required: @Initialized @NonNull String
```

Original issue reported on code.google.com by `cus...@google.com` on 5 Jan 2015 at 6:39
